seacrowd/sea_datasets/unisent/unisent.py

```diff
@@ -0,0 +1,142 @@
+    "mwv", "mya", "nbe", "nfa", "nia", "nij", "nlc", "ntx", "npy", "obo",
+    "pag", "pam", "plw", "pmf", "pne", "ppk", "prf", "prk", "ptu", "pww",
+    "sas", "sbl", "sda", "sgb", "smk", "sml", "sun", "sxn", "szb", "tbl",
+    "tby", "tcz", "tdt", "tgl", "tha", "tih", "tlb", "twu", "urk", "vie",
+    "war", "whk", "wrs", "xbr", "yli", "yva", "zom", "zlm", "zyp", "pse",
+    "mnx", "mmn", "lsi", "hlt", "gdg", "bnj", "acn",
+]
+
+_LICENSE = Licenses.CC_BY_NC_ND_4_0.value  # cc-by-nc-nd-4.0
+_LOCAL = False
+_URLS = {
+    _DATASETNAME: {lang: URL.format(lang) for lang in _LANGUAGES},
+}
+
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class UniSentDataset(datasets.GeneratorBasedBuilder):
+    LABELS = ["NEGATIVE", "POSITIVE"]
+
+    BUILDER_CONFIGS = ([
+                           SEACrowdConfig(name=f"{_DATASETNAME}_{lang}_source",
+                                          version=datasets.Version(_SOURCE_VERSION),
+                                          description=_DESCRIPTION,
+                                          schema="source",
+                                          subset_id=f"{_DATASETNAME}_{lang}", ) for lang in _LANGUAGES
+                       ]
+                       +
+                       [
+                           SEACrowdConfig(
+                               name=f"{_DATASETNAME}_{lang}_seacrowd_text",
+                               version=datasets.Version(_SEACROWD_VERSION),
+                               description=_DESCRIPTION,
+                               schema="seacrowd_text",
+                               subset_id=f"{_DATASETNAME}_{lang}",
+                           )
+                           for lang in _LANGUAGES
+                       ])
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        features = None
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "word": datasets.Value("string"),
+                    "lexicon": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "seacrowd_text":
+            features = schemas.text_features(label_names=self.LABELS)
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        lang = self.config.subset_id.split("_")[-1]
+        urls = _URLS[_DATASETNAME][lang]
+        data_dir = dl_manager.download_and_extract(urls)
+        return [
+            datasets.SplitGenerator(
+                name="unisent",
+                gen_kwargs={
+                    "filepath": data_dir,
+                },
+            ),
```

Comment:
It should be like this:

```
datasets.SplitGenerator(
    name=datasets.Split.TRAIN,
    gen_kwargs={
        "filepath": os.path.join(data_dir),
        "split": "train",
    },
)
```

Reply:
I don't think I have to do this as I didn't use the `split` args